src/core1/DynamicGenericArray.java

```diff
@@ -0,0 +1,51 @@
+package core1;
+
+public class DynamicGenericArray <T> {
+    private Object[] array;
+    private int currLength;
+
+    public DynamicGenericArray(int length) {
+        array = new Object[length];
+        currLength = 0;
+    }
+
+    public DynamicGenericArray() {
+        this(10);
```

Comment:
I understand why here is a "10" value and what it means but it's better to create constant variable with talking name.